javascript/code-challenges/challenges-03.test.js

```diff
@@ -0,0 +1,360 @@
+"use strict";
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function called addTwo that takes in an array and adds two to every value using a for loop. Place the new value in a new array. Return the new array.
+------------------------------------------------------------------------------------------------ */
+
+const addTwo = (arr) => {
+  const newArr = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    const value = arr[i];
+    newArr.push(value + 2);
+  }
+  return newArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named typeNum that, given an array as input, uses filter to return an array containing only the numbers.
+
+For example, typeNum([1, 'bob' ,3]) returns [1,3].
+------------------------------------------------------------------------------------------------ */
+
+const typeNum = (arr) => {
+  const numbersOlny = arr.filter(Number);
+  return numbersOlny;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named containsAnd that, given an array of strings as input, uses filter to return an array containing only strings that contain 'and' within the string.
+
+For example, containsAnd(['panda', 'ran', 'and']) returns ['panda', 'and'].
+------------------------------------------------------------------------------------------------ */
+
+const containsAnd = (arr) => {
+  let word = "and";
+  const onlyAnd = arr.filter((str) => str.includes(word));
+
+  return onlyAnd;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named oddValues that, given an array of integers as input, uses filter to return an array containing only the odd integers.
+
+For example, oddValues([1,2,3]) returns [1,3].
+------------------------------------------------------------------------------------------------ */
+
+const oddValues = (arr) => {
+  const evenNums = arr.filter((num) => !((num % 2) - 1));
+  return evenNums;
```

Comment:
In this case you can simple return
```js
return arr.filter(num => num % 2 !== 0);
```
Which, then you can turn intos a single liner
```js
const oddValues = arr => arr.filter(num => num % 2 !== 0);
```